docker/lwua-ingest/lwua-py/lwua/graphdb.py

```diff
@@ -0,0 +1,251 @@
+# this prevents the injest.py and tryout_watch.py to be run at the same
+# time with conflicting circular imports
+
+from SPARQLWrapper import SPARQLWrapper, JSON
+from datetime import datetime
+from pathlib import Path
+from urllib.parse import unquote, quote
+import logging
+from rdflib import Graph
+import time
+import os
+
+# from dotenv import load_dotenv
+from .helpers import resolve_path  # ,enable_logging
+from pyrdfj2 import J2RDFSyntaxBuilder
+
+log = logging.getLogger(__name__)
+URN_BASE = os.getenv("URN_BASE", "urn:lwua:INGEST")
+
+
+def gdb_from_config():
+    base = os.getenv("GDB_BASE", "http://localhost:7200")
+    repoid = os.getenv("GDB_REPO", "lwua23")
+
+    endpoint = f"{ base }/repositories/{ repoid }"
+    # update statements are handled at other endpoint
+    updateEndpoint = endpoint + "/statements"
+
+    log.debug(f"using endpoint {endpoint}")
+
+    GDB = SPARQLWrapper(
+        endpoint=endpoint,
+        updateEndpoint=updateEndpoint,
+        returnFormat=JSON,
+        agent="lwua-python-sparql-client",
+    )
+    GDB.method = "POST"
+    return GDB
+
+
+GDB = gdb_from_config()
+
+
+def get_j2rdf_builder():
+    template_folder = resolve_path("./lwua/templates")
+    log.info(f"template_folder == {template_folder}")
+    # init J2RDFSyntaxBuilder
+    context = f"{URN_BASE}:ADMIN"
+    j2rdf = J2RDFSyntaxBuilder(
+        templates_folder=template_folder,
+        extra_functions={"registry_of_lastmod_context": context},
+    )
+    return j2rdf
+
+
+J2RDF = get_j2rdf_builder()
+
+
+def update_registry_lastmod(context: str, lastmod: datetime):
+    """
+    Update the administration of a context.
+
+    :param context: The context to update.
+    :type context: str
+    :param lastmod: The date string to update with.
+    :type lastmod: str
+    """
+    log.info(f"update registry_of_lastmod_context on {context}")
+
+    # check if context is IRI compliant
+    assert_iri_compliance(context)
+
+    # variables for the template
+    template = "update_context_lastmod.sparql"
+    vars = {
+        "context": context,
+        "lastmod": lastmod if lastmod is not None else None,
+    }
+    # get the sparql query
+    query = J2RDF.build_syntax(template, **vars)
+    log.debug(f"update_registry_lastmod query == {query}")
+    # execute the query
+    GDB.setQuery(query)
+    GDB.query()
+
+
+def assert_iri_compliance(context: str):
+    assert context.startswith(
+        URN_BASE), f"Context {context} is not IRI compliant"
+
+
+def insert_graph(graph: Graph, context: str = None):
+    """
+    Insert data into a context.
+
+    :param graph: The graph to insert data from.
+    :type graph: Graph
+    :param context: The context to insert data into.
+    :type context: str
+    """
+
+    log.info(f"insert_graph into {context}")
+    assert_context_exists(context)
+    # Initialize the J2RDFSyntaxBuilder
+
+    assert_iri_compliance(context) if context is not None else None
```

Comment:
drop the remainder if on this line once the assert_uri_compliance has learned to survive context=None